pallets/collator-selection/src/lib.rs

```diff
@@ -377,11 +377,11 @@ pub mod pallet {
 		pallet_authorship::EventHandler<T::AccountId, T::BlockNumber> for Pallet<T>
 	{
 		fn note_author(author: T::AccountId) {
-			let treasury = Self::account_id();
-			let reward = T::Currency::free_balance(&treasury).div(2u32.into());
-
-			// `reward` is half of treasury account, this should never fail.
-			let _success = T::Currency::transfer(&treasury, &author, reward, KeepAlive);
+			let pot = Self::account_id();
+			// assumes an ED will be sent to pot.
+			let reward = T::Currency::free_balance(&pot).checked_sub(&T::Currency::minimum_balance()).unwrap_or_else(Zero::zero).div(2u32.into());
```

Comment:
looks a bit strange to me. Now, even if the pot is very large, we are subtracting `T::Currency::minimum_balance()` from it before dividing by two. Why? 

Better would be to use the `Currency` API: 

```
let pot = Self::account_id();
// amount that we want to reward to collators 
let to_reward = T::Currency::free_balance(&pot).div(2u32.into());
if T::Currency::can_withdraw(pot, to_reward).is_ok() {
     // tranfer `to_reward` to `author`
} else {
     // the pot is not large enough. Don't do anything yet.
}

Reply:
We don't expect the pot to get large since it is cut in half each block and would take several hundred txs just to accumulate the ED in fees.

Reply:
Or since we do `KeepAlive`, it is okay as it is. But you comment saying

```
// `reward` is half of pot account, this should never fail.
``` 

is wrong. This could very well fail. On the first blocks, the pot is almost equal to `ED + epsilon`. Half of this is well below `ED` and will cause the pot to become dust, and therefore this will fail.

Reply:
So we are aware that this is not uniform? For many blocks, the pot's size will be so small that half of it is not large enough to be transferred out. Then suddenly, when the pot is `2 * ED`, we give half of it to one collator, and repeat? 

I think this is kinda unfair and probably collators will be only interested to author that block that makes the pot become `2 * ED`.

Reply:
I don't think it is that way, I see it as If ED is 5. We add 5 ourselves to the pot, now pot is above ED, first rewards come in let's ssay it is 2. 7 - 5 / 2 the reward would be one.

Reply:
indeed, I was wrong and missed the fact that we subtract ED from the pot and then divide it. 👍